mypy/checkexpr.py

```diff
@@ -2206,6 +2210,22 @@ def visit_comparison_expr(self, e: ComparisonExpr) -> Type:
         assert result is not None
         return result
 
+    def find_partial_type_ref_fast_path(self, expr: Expression) -> Optional[Type]:
+        """If expression has a partial generic type, return it without additional checks.
+
+        In particular, this does not generate an error about a missing annotation.
+
+        Otherwise, return None.
+        """
+        if not isinstance(expr, RefExpr):
+            return None
+        if isinstance(expr.node, Var):
+            result = self.analyze_var_ref(expr.node, expr)
+            if isinstance(result, PartialType) and result.type is not None:
+                self.chk.store_type(expr, result)
```

Comment:
TBH I am comfortable with putting some extra partial types in type map. A while ago we had a bunch of crashes caused by this. Should we instead use `fixup_partial_type()` here?

Reply:
Yeah, probably! I will try it.